### PR TITLE
Mnt/generalize plot varargs

### PR DESCRIPTION
**Pull request by tacaswell**
## PR summary

This extends the `_plot_var_args` helper class slightly so that it can be used in the data-prototype work.  The key idea here is that in addition to being able to return instantiated Artists, it can also return the inferred x, y, and kwargs. 

This is all private API.

## PR checklist

- [/] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [/] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [/] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [/] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [/] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by ksunden — APPROVED** [review]

File: lib/matplotlib/axes/_base.py
Comment (tacaswell) [reply lib/matplotlib/axes/_base.py:217]: We wanted to be able to re-use all of the machinery for processing the vargs, but get the per-artist args/wkwargs out so the we can create new-style artist from them (which we can not inject into core Matplotlib yet).

I agree it is not great, but was going for the minimal change that would unblock the data prototype work without having to re-write this whole API!

Instead of `"mirror"` use `"no_artist"` instead?

Comment (timhoffm) [reply lib/matplotlib/axes/_base.py:217]: This is all internal, so renaming would be possible. However there's a slight complication that `command` is used in `raise _api.kwarg_error(self.command, pos_only)`. With that we inject the usage context name into the class, so that when called from `plot()` it can issue an error containing "plot" and likewise for "fill". This means that `_process_plot_var_args` is not really context free and you'd have to hard-code in which places you want to use it to support a proper exception.

The "cheap" solution would be `command="UNKNOWN"` or similar, which would result in

> UNKNOWN() got an unexpected keyword argument 'x'

That's sort of ok.

The IMHO slightly better would be to decouple the function from the caller, i.e. return a generic "unexpected keyword arguments 'x'" and - if desired - catch and ammend that in the caller. This would give the freedom to rename `command` - e.g. to `artist` with values Line2D, Polygon and None.

**Comment by tacaswell** [comment]
I am 👍🏻 on @timhoffm changes.

**Review by timhoffm — APPROVED** [review]
I approve. I suggest to Tom and me together as one reviewer, covering each others contribution. So if @ksunden agrees with the changes done after his approval, we can merge.

**Review by ksunden — APPROVED** [review]

**Pull request merged**
